src/app/copy/inline-copy/inline-copy.component.html

```diff
@@ -0,0 +1,16 @@
+<span
+  class="inline-copy">
+  <span
+    class="token-cont"
+    data-toggle="tooltip"
+    data-placement="top"
+    [tooltip]="label">
+    {{token}}
+  </span>
+  <button
+    class="copy-btn"
+    [attr.aria-label]="copyBtnTxt + ' ' + label"
```

Comment:
Yea I was definitely noticing some limitations of this approach when I was writing it. That said, it did have the nice effect of 1) ensuring a sensible, and unique accessible name was generated for each control and 2) we know that the experience will be uniform for this type of control to the screen reader user.

I see though why it's good not to make assumptions about how users of the framework will want to express their content to their readers. Maybe we could make this a setting, that's set by default to ensure a baseline level of accessibility is provided by the framework, and then the ability to turn it off and use your own aria attributes if you want?

For now i'll focus on using ariaLabel as an input as is and we can think about it from there.

Reply:
Now that I think about it might be good to just show this as an example or something. I'll come up with something more fleshed out and we can go from there